README.org

```diff
@@ -371,19 +373,163 @@ We can also change the thickness of the lines and shapes that we draw by changin
+   (translate 100 100)
+   (dotimes (x 5)
+     (rect 0 0 100 100)
+     (translate 150 0)
+	 (scale 1/2)))
+#+END_SRC
 
 *** Text
-TODO
+Use =(text text-string x y &optional width height)= to draw text, where =x= and =y= specify the top-left corner of the rectangle containing the text. =width= and =height= control the shape of the text box. There is support for changing the [[https://github.com/vydd/sketch/blob/master/src/font.lisp#L29][font]].
+
+#+BEGIN_SRC lisp
+  (defsketch text-test
+     ((title "Hello, world!"))
+   (text title 0 0 100))
+#+END_SRC
 
 *** Images
-TODO
+First =(load-resource filename ...)= to load the image from a given file, then =(image image-resource x y &optional width height)= to draw the image with its top-left corner at =(x, y)= and with the given =width= and =height=.
+
+#+BEGIN_SRC lisp
+  (defsketch image-test
+     ((title "Hello, image!")
+      (rsc (load-resource "/path/to/img.png")))
+    (image rsc 10 10 200 200))
+#+END_SRC
 
-*** Bindings
-TODO
+Images can be cropped using =(crop (image-resource image) x y w h)=, where =x= and =y= indicate the top-left corner of the cropping rectangle and =w= and =h= indicate the width & height. Image flipping can be accomplished by using negative =w= and =h= values.
 
 *** Input
-TODO
+Input is handled by overriding methods provided by [[https://github.com/lispgames/sdl2kit/#Windows][sdl2kit]]; the sketch window is a subclass of the =WINDOW= class provided by =sdl2kit=, which provides these generic functions for handling input. Drawing functions cannot be called from these methods, only from the body of =defsketch=.
+
+In this example, we draw a new rectangle every time there is a click. The important parameters of the event method: =state= is a keyword symbol indicating the state of the mouse, =ts= is a timestamp, =x= and =y= are the coordinates of the mouse click.
+
+#+BEGIN_SRC lisp
+  (defsketch input-test
+     ((title "Hello, input")
+      (rectangles nil))
+   (loop for (x y) in rectangles
+         do (rect x y 50 50)))
+  (defmethod kit.sdl2:mousebutton-event ((window input-test) state ts b x y)
+    (with-slots (rectangles) window
+      (when (eq state :mousebuttondown)
+        (push (list x y) rectangles))))
+#+END_SRC
+
+In this example, all keyboard text input is echoed to the screen. =text= is a string / character array containing a single character.
+
+#+BEGIN_SRC lisp
+  (defsketch input-test
+     ((title "Hello, input")
+      (text-to-write nil))
+    (loop for s in text-to-write
+          do (text s 0 0 20 20)
+          do (translate 20 0)))
+  (defmethod kit.sdl2:textinput-event ((window input-test) ts text)
+    (with-slots (text-to-write) window
+      (setf text-to-write (nconc text-to-write (list text)))))
+#+END_SRC
+
+Finally, here is an example where a pair of eyes follow the mouse (the pupils are restricted to a rectangle, it would look better if they were restricted to a circle).
+
+#+BEGIN_SRC lisp
+  (defsketch input-test
```

Comment:
Consider using different names for different examples?
It can be surprising if someone defines the previous sketch, then this one, then clicks/types and an error occurs because the previous method wasn't "invalidated".